spring-boot-starter/autoconfigure/src/main/java/tech/cassandre/trading/bot/ta4j/BarContext.java

```diff
@@ -70,27 +80,73 @@
         this.startTime = newStartTime;
         this.endTime = startTime.plus(duration);
         this.close = newClose != null ? newClose.doubleValue() : 0;
-
-        this.low = newLow != null ? newLow.doubleValue() : close;
-        this.high = newHigh != null ? newHigh.doubleValue() : close;
         this.open = newOpen != null ? newOpen.doubleValue() : close;
-        this.volume = newVolume != null ? newVolume.doubleValue() : 0;
+
+        this.low = Math.min(close, open);
+        this.high = Math.max(close, open);
+
+
+        calculateInitialVolume(newVolume);
+    }
+
+    private void calculateInitialVolume(final Number newVolume) {
+        if (newVolume == null) {
+            return;
+        }
+        if (isDurationMoreThanDay()) {
+            this.volume = newVolume.doubleValue();
+        } else {
+            this.initialDayVolume = newVolume.doubleValue();
+        }
     }
 
     public boolean isAfter(final ZonedDateTime timestamp) {
         return timestamp.isAfter(endTime.minus(Duration.ofSeconds(1)));
     }
 
+    /**
+     * The contract of the update call is that it is called without timestamp
+     * and is always within bounds of one bar duration.
+     *
+     * @param newClose  new close - mandatory
+     * @param newVolume new volume to be updated (might be null)
+     */
     @SuppressWarnings("checkstyle:AvoidInlineConditionals")
-    public void update(final Number newLow, final Number newHigh, final Number newClose, final Number newVolume) {
+    public void update(final Number newClose, final Number newVolume) {
         if (newClose == null) {
             throw new IllegalArgumentException("Cannot update bar context without at least specifying close price");
         }
         close = newClose.doubleValue();
-        low = Math.min(low, newLow == null ? close : newLow.doubleValue());
-        high = Math.max(high, newHigh == null ? close : newHigh.doubleValue());
+        low = Math.min(low, close);
```

Comment:
I'm not sure if that will work.
If we would receive all trades and updates from the exchange, then yes. But we get it only every X seconds (configurable).

Example:

Exchange: trades (**low** marked fat): 3,  **2**,  4,  **2**,  5,  **2**
Cassandre close-->low calculation (if we only receive every second tick): **3**,  4-->**3**,  5-->**3**

As we dont recieve all updates from the exchange, we would never get the actual low of **2**

(same is true of course for high)

Solution idea:
Similar to the volume, we store the initial low and initial high in a dummy variable.
If the current low is lower than that initial seen low, we update the low price (same for high)